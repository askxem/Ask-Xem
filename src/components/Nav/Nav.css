header {
    position: relative;
    z-index: 10;
    padding: 10px;
    margin-bottom: 20px;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    align-items:center;
    justify-content: space-between;
    box-shadow: 0px 5px 10px gray;
    background-color: white;
    gap: 10px;
}

nav img {
    width: 50px;
    cursor: pointer;
    background-color: #ecb9a1;
    border-radius: 100%;
    padding: 10px;
    box-shadow: 0px 5px 10px gray;
}

.navicons {
    display: flex;
    gap: 15px;
}

header span {
    margin-right: 5px;
    font-size: .9em;
}

header button {
    border: none;
    background-color: #91ad8c;
    border-radius: 5px;
    padding: 3px;
    color: white;
    cursor: pointer;
}

header button:hover, button:focus {
    background-color: #E57C49;
}

@media only screen and (min-width: 768px) {
    nav img {
        width: 80px;
    }
}