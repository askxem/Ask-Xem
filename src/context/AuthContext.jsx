import { createContext, useState, useContext, useMemo } from "react";
import { getUser, signInUser, signOutUser, signUpUser } from "../services/users.js";


const AuthContext = createContext();

function ProvideAuth({children}) {

    const currentUser = getUser();

    const [user, setUser] = useState(
        currentUser ? {...currentUser} : {}
    );

    async function signUp(email, password) {
        const newUser = await signUpUser(email, password);
        setUser(newUser);
    }

    async function signIn(email, password) {
        const newUser = await signInUser(email, password);
        setUser(newUser);
    }

    async function signOut() {
        await signOutUser();
        localStorage.clear();
        setUser({});
        history.push('/home');
    }

    const value = useMemo(() => ({ user, signUp, signIn, signOut}), [user]);

    return (
        <AuthContext.Provider value={value} >
            {children}
        </AuthContext.Provider>
    )
}

/**
 * 
 * @returns a user object, a signup, signin, and signout fn that hits the database.
 */
function useAuth(){
    const context = useContext(AuthContext);

    if (context === undefined) throw new Error('Auth context not accessible outside of the auth provider.')

    return context;
}


export { ProvideAuth, useAuth }