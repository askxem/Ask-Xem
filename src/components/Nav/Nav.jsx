import React, { useState } from 'react';
import { BrowserRouter, Link } from 'react-router-dom';
import { NavLink, Router } from 'react-router-dom';
import { useAuth } from '../../context/AuthContext.jsx';
import Journal from '../Journal/Journal.jsx';

export default function Nav() {
    const [showJournal, setShowJournal] = useState(false);
    const {user} = useAuth();

  return (
        //Nav links here!
        <nav>
          <Link to='/select'>Home Img</Link>
          <Link to='/favorites'>Heart Img</Link>
          <button onClick={() => setShowJournal(true)} disabled={!user.id}>Journal</button>
          <Journal showJournal={showJournal} setShowJournal={setShowJournal}/>
        </nav>
  );
}
