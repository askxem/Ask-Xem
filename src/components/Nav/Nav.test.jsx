import {render} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ProvideAuth } from '../../context/AuthContext.jsx';
import Nav from './Nav.jsx'

jest.mock('../../context/AuthContext.jsx');
jest.mock('../../services/users.js');
jest.mock("../../services/journal.js");

describe('renders nav component', () => {
    it('render nav bar without user in context', () => {
        const {container} = render(
            <ProvideAuth>
                <MemoryRouter>
                    <Nav />
                </MemoryRouter>
            </ProvideAuth>
        )
    
        expect(container).toMatchSnapshot();
    });

    it('render nav bar with user in context', () => {
        const {container} = render(
            <ProvideAuth mockUser={{id: 2, email: 'test@test.com'}}>
                <MemoryRouter>
                    <Nav />
                </MemoryRouter>
            </ProvideAuth>
        )
    
        expect(container).toMatchSnapshot();
    });
})