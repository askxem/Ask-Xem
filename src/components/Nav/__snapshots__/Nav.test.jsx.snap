// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nav component render nav bar with user in context 1`] = `
<div>
  <nav>
    <a
      href="/select"
    >
      Home Img
    </a>
    <a
      href="/favorites"
    >
      Heart Img
    </a>
    <button>
      Journal
    </button>
  </nav>
  <article>
    <p>
      Signed in as 
      test@test.com
    </p>
    <button>
      Signout
    </button>
  </article>
</div>
`;

exports[`renders nav component render nav bar without user in context 1`] = `
<div>
  <article>
    <a
      href="/login"
    >
      Login
    </a>
  </article>
</div>
`;
